core/src/main/scala/org/apache/spark/sql/crossdata/catalog/CatalogChain.scala

```diff
@@ -125,16 +125,30 @@ private[crossdata] class CatalogChain private(val temporaryCatalogs: Seq[XDTempo
   override def persistView(tableIdentifier: ViewIdentifier, plan: LogicalPlan, sqlText: String): Unit =
     persistentCatalogs.foreach(_.saveView(tableIdentifier, plan, sqlText))
 
+  override def persistIndex(crossdataIndex: CrossdataIndex): Unit =
+    if (persistentChainedLookup(_.lookupTable(crossdataIndex.tableIdentifier)).isEmpty) {
+      throw new RuntimeException(s"Cannot create the index. Table ${crossdataIndex.tableIdentifier} doesn't exist or is temporary")
+    } else {
+      persistentCatalogs.foreach(_.saveIndex(crossdataIndex))
+    }
+
+
   override def dropTable(tableIdentifier: TableIdentifier): Unit = {
     val strTable = tableIdentifier.unquotedString
     if (!tableExists(tableIdentifier)) throw new RuntimeException(s"Table $strTable can't be deleted because it doesn't exist")
     logInfo(s"Deleting table $strTable from catalog")
+
+    indexMetadataByTableIdentifier(tableIdentifier) map { index =>
```

Comment:
Use `foreach` instead of `map` as you are interested just in the `dropIndex` side effect and not its return value.